### ISSUE TITLE
`truncate` command for dates

### ISSUE THREAD

**Issue by mlucy** [rethinkdb/rethinkdb#3710]
@coffeemug asked for this and can maybe give more information.

**Comment by coffeemug** [comment]
Currently I can run:

```js
> r.now().day()
2

> r.now().month()
2

// etc.
```

I want to be able to do:

```js
> r.now().day({prefix: true})
"2015-02-02"

> r.now().month({prefix: true})
"2015-02"

// etc.
```

This would make it easier to group by month or day, whereas right now I would have to construct the group manually and it's enough of a pain to do it that RethinkDB isn't a great tool for such operations.

**Comment by coffeemug** [comment]
Also note, the above is meant to illustrate what I want to do, not to dictate the API. We might want to come up with a different API to do it (for example, do we want to return strings or something else?), but I don't understand dates and times well enough to propose something sensible.

**Comment by marshall007** [comment]
I'd much rather see a command that allows arbitrary formatting, something like:

```js
> r.now().format("YYYY-MM-DD")
"2015-02-02"

> r.now().format("YYYY-MM")
"2015-02"
```

**Comment by VeXocide** [comment]
`date` has a nice format that might work, http://man7.org/linux/man-pages/man1/date.1.html.

**Comment by coffeemug** [comment]
I *really* like the `format` command.

One concern I have with it is that whenever I format things like this I always find myself digging through documentation to find a correct string to represent what I want, which always annoys me. It doesn't seem very... conducive to quick and dirty time series grouping queries, though it might be a matter of good docs.

**Comment by VeXocide** [comment]
For the record, this is probably related to #3353 and #3387.

**Comment by danielmewes** [comment]
I think I would prefer a `format` term over having a `prefix` opt arg.
Even though you might have to look up the placeholders to use in the format string, it is much more flexible and has a lot of other use cases as well.

**Comment by marshall007** [comment]
> @coffeemug: One concern I have with it is that whenever I format things like this I always find myself digging through documentation to find a correct string to represent what I want, which always annoys me.

Agreed, I think one solution would be to allow passing a "human-readable" unit (i.e. `year`, `month`, `day`, etc) which would give you an ISO 8601 formatted date up to and including the unit specified.

So `.format('day')` would just be an alias to `.format('YYYY-MM-DD')`.

**Comment by timmaxw** [comment]
I think that `YYYY-MM-DD` is an easy enough format string to remember. The problem that I usually have with date format strings is that they look like `%Y-%m-%d`, not to be confused with `%y-%m-%d`, `%Y-%M-%d`, or `%Y-%m-%D`, all of which are valid format strings that do different things. If we design our format syntax so that `YYYY-MM-DD` literally works and does what it looks like, I think it would be OK.
